src/coreComponents/dataRepository/Utilities.cpp

```diff
@@ -0,0 +1,183 @@
+  }
+
+  MpiWrapper::gather( localAllocations.data(),
+                      numValues,
+                      globalAllocations.data(),
+                      numValues,
+                      0,
+                      MPI_COMM_GEOSX );
+
+  // reduce data across ranks (min, max, sum)
+  if( MpiWrapper::commRank()==0 )
+  {
+    array2d< double > allocationReductions( numValues, 3 );
+    for( int a=0; a<numValues; ++a )
+    {
+      allocationReductions( a, 0 ) = 1e99;
+      allocationReductions( a, 1 ) = 0;
+      allocationReductions( a, 1 ) = 0;
+      for( int b=0; b<numRanks; ++b )
+      {
+        int const recvIndex = a + b * numValues;
+        double const value = globalAllocations[recvIndex];
+        allocationReductions( a, 0 ) = std::min( allocationReductions( a, 0 ), value );
+        allocationReductions( a, 1 ) = std::max( allocationReductions( a, 1 ), value );
+        allocationReductions( a, 2 ) += value;
+      }
+    }
+
+    // scope for the output of groups
+    {
+      double const * const groupAllocations = allocationReductions[0];
+
+      if( indent==0 )
+      {
+        //                          1         2         3         4         5         6         7         8         9        10        11
+        //        12
+        //                 123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890
+        GEOSX_LOG_RANK_0( "************************************************************************************************************************" );
+        GEOSX_LOG_RANK_0( " Data repository memory allocations " );
+        GEOSX_LOG_RANK_0( "                                                                                       min          max          sum    " );
+        GEOSX_LOG_RANK_0( "                                                                                   -----------  -----------  -----------" );
+      }
+
+
+      string outputLine;
+      int indentChars = 0;
+      for( int i=0; i<indent; ++i )
+      {
+        outputLine += terminateBranch[i]==true ? "   " : "|  ";
+        indentChars += 3;
+      }
+      // put a indent between groups in the tree
+      GEOSX_LOG_RANK_0( outputLine.c_str()<<"|" );
+      indentChars += 1;
+      // only allocation data if it is above the threshold...
+      if( groupAllocations[0] >= threshold )
+      {
+        indentChars += 3;
+        outputLine += "|--{:.<" + std::to_string( 83-indentChars ) + "} {:>9s}    {:>9s}    {:>9s}";
+        GEOSX_LOG_RANK_0( GEOSX_FMT( outputLine.c_str(),
+                                     "[" + group.getName() + "]",
+                                     stringutilities::toMetricPrefixString( groupAllocations[0] ) + 'B',
+                                     stringutilities::toMetricPrefixString( groupAllocations[1] ) + 'B',
+                                     stringutilities::toMetricPrefixString( groupAllocations[2] ) + 'B' ) );
```

Comment:
You could also use a nested width arg instead of manually putting the width specifier into the format string:
```suggestion
        outputLine += "|--{:.<{}} {:>9s}    {:>9s}    {:>9s}";
        GEOSX_LOG_RANK_0( GEOSX_FMT( outputLine.c_str(),
                                     83-indentChars,
                                     "[" + group.getName() + "]",
                                     stringutilities::toMetricPrefixString( groupAllocations[0] ) + 'B',
                                     stringutilities::toMetricPrefixString( groupAllocations[1] ) + 'B',
                                     stringutilities::toMetricPrefixString( groupAllocations[2] ) + 'B' ) );
```

Reply:
I tried this suggestion, but I got this error:
```
|--[Problem]
   |
   |--[domain]
   |  |
   |  |--[MeshBodies]
   |  |  |
   |  |  |--[mesh1]
   |  |     |
   |  |     |--[meshLevels]
   |  |        |
   |  |        |--[Level0]
   |  |        |  |
width is not integer
--------------------------------------------------------------------------
MPI_ABORT was invoked on rank 0 in communicator MPI_COMM_WORLD
with errorcode 1.

NOTE: invoking MPI_ABORT causes Open MPI to kill all MPI processes.
You may or may not see output from other processes, depending on
exactly when Open MPI kills them.
```
any ideas?